src/gatherers/critical-network-chains.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Gather = require('./gather');
+const log = require('../lib/log.js');
+
+const flatten = arr => arr.reduce((a, b) => a.concat(b), []);
+const includes = (arr, elm) => arr.indexOf(elm) > -1;
+
+class Node {
```

Comment:
I like RequestNode better, because it holds data about a request. The initiator relationship is captured by the edges, so I'll add a comment about what the parent-child relationship represents.